### DISCUSSION TITLE
Running EVAC on a FDS scenario which has MPI Process enabled

### DISCUSSION THREAD

**Discussion by LordMicidial** [firemodels/fds#13498] [Q&A]
Hi there,

I've happily finished my first FDS simulation few months ago. Even if I might have been too much scrupulous, I ended up with a a file which consist of 24 MESHes to represent a logistic warehouse where a fire start and as soon as Temperature get higher than sprinklers activation temperature the HRR curve get stuck in order to represent the mitigation effect of the fire fighting system as per local legislation.

The warehouse is about  140 m x 85 m with an height of approx 9 m at the edge of the roof. I report here below the meshes i'd used in the final simulation.

&MESH ID='Mesh01-01', IJK=120,110,55, XB=-1.0,23.0,-1.0,21.0,-1.0,10.0, MPI_PROCESS=0/
&MESH ID='Mesh01-02', IJK=120,110,55, XB=-1.0,23.0,21.0,43.0,-1.0,10.0, MPI_PROCESS=1/
&MESH ID='Mesh01-03', IJK=120,110,55, XB=-1.0,23.0,43.0,65.0,-1.0,10.0, MPI_PROCESS=2/
&MESH ID='Mesh01-04', IJK=120,110,55, XB=-1.0,23.0,65.0,87.0,-1.0,10.0, MPI_PROCESS=3/
&MESH ID='Mesh01-05', IJK=120,110,55, XB=23.0,47.0,-1.0,21.0,-1.0,10.0, MPI_PROCESS=4/
&MESH ID='Mesh01-06', IJK=120,110,55, XB=23.0,47.0,21.0,43.0,-1.0,10.0, MPI_PROCESS=5/
&MESH ID='Mesh01-07', IJK=120,110,55, XB=23.0,47.0,43.0,65.0,-1.0,10.0, MPI_PROCESS=6/
&MESH ID='Mesh01-08', IJK=120,110,55, XB=23.0,47.0,65.0,87.0,-1.0,10.0, MPI_PROCESS=7/
&MESH ID='Mesh01-09', IJK=120,110,55, XB=47.0,71.0,-1.0,21.0,-1.0,10.0, MPI_PROCESS=8/
&MESH ID='Mesh01-10', IJK=120,110,55, XB=47.0,71.0,21.0,43.0,-1.0,10.0, MPI_PROCESS=9/
&MESH ID='Mesh01-11', IJK=120,110,55, XB=47.0,71.0,43.0,65.0,-1.0,10.0, MPI_PROCESS=10/
&MESH ID='Mesh01-12', IJK=120,110,55, XB=47.0,71.0,65.0,87.0,-1.0,10.0, MPI_PROCESS=11/
&MESH ID='Mesh01-13', IJK=120,110,55, XB=71.0,95.0,-1.0,21.0,-1.0,10.0, MPI_PROCESS=12/
&MESH ID='Mesh01-14', IJK=120,110,55, XB=71.0,95.0,21.0,43.0,-1.0,10.0, MPI_PROCESS=13/
&MESH ID='Mesh01-15', IJK=120,110,55, XB=71.0,95.0,43.0,65.0,-1.0,10.0, MPI_PROCESS=14/
&MESH ID='Mesh01-16', IJK=120,110,55, XB=71.0,95.0,65.0,87.0,-1.0,10.0, MPI_PROCESS=15/
&MESH ID='Mesh01-17', IJK=120,110,55, XB=95.0,119.0,-1.0,21.0,-1.0,10.0, MPI_PROCESS=16/
&MESH ID='Mesh01-18', IJK=120,110,55, XB=95.0,119.0,21.0,43.0,-1.0,10.0, MPI_PROCESS=17/
&MESH ID='Mesh01-19', IJK=120,110,55, XB=95.0,119.0,43.0,65.0,-1.0,10.0, MPI_PROCESS=18/
&MESH ID='Mesh01-20', IJK=120,110,55, XB=95.0,119.0,65.0,87.0,-1.0,10.0, MPI_PROCESS=19/
&MESH ID='Mesh01-21', IJK=120,110,55, XB=119.0,143.0,-1.0,21.0,-1.0,10.0, MPI_PROCESS=20/
&MESH ID='Mesh01-22', IJK=120,110,55, XB=119.0,143.0,21.0,43.0,-1.0,10.0, MPI_PROCESS=21/
&MESH ID='Mesh01-23', IJK=120,110,55, XB=119.0,143.0,43.0,65.0,-1.0,10.0, MPI_PROCESS=22/
&MESH ID='Mesh01-24', IJK=120,110,55, XB=119.0,143.0,65.0,87.0,-1.0,10.0, MPI_PROCESS=23/
 
Now, I need to re-start the same simulation, this time with the EVAC Module because the local fire brigate want me to prove the safe evacuation. 

In order to do so, I got into the topic of EVAC and studied the manual. I actually managed to run a simulation with a unique FDS MESH encompassing all of the above MESHES which worked fine and speedy due to the very low level of HRRPUA and an EVAC MESH with the same IJK and coordinates, z offset=1.5.

&MESH ID='Mesh01-01', IJK=120,110,55, XB=-1.0,143.0,-1.0,89.0,-1.0,10.0/
&MESH ID='Evac_Mesh01', IJK=120,110,1, XB=-1.0,143.0,-1.0,89.0,0.0,3.0, EVAC_Z_OFFSET= 1.500,  EVACUATION=.TRUE.,  EVAC_HUMANS=.TRUE./

(89 on the Y's instead of the above 87s make sense, at least to me...)

While comparing the simulations, it seems that the spreading of the SOOT differs a little and overall the 24 MESH is for sure better represented in term of OBSTs if compared with the EVAC scenario has more grainy and imperfect geometries. Also, to my great sorrow, i just noticed that the output of the hrr.csv files differ too much.

So I'm trying to run the original simulation (24 FDS MESH) superimposing an EVAC Mesh. My PC is (i9, with 24 physical core and 32 threads).

FDS start to compile the file and manage to produce the .smv scenario which resulted grainy and imperfect as per the first EVAC scenario with the unique FDS MESH. 

However this time I got a memory allocation issue as per the screenshot below.

What do you think I must do or learn to fix the problem?
I've run the come typing: mpiexec -np 24 fds namefile.fds 

I've seen online tutorial mentioning OMP_NUM_THREADS=N_THREADS_MP as a tool to change the number of thread of the simulation. The code seems not working, at least to me. However, given the fact that EVAC cannot run on MPI, threads should be the way. The User Guide doesn't really explain how to play with threads to best scale FDS+EVAC unfortunately. 

I hope I can find help here even if I have not understood if EVAC is still on development.

Kind regards!

![image](https://github.com/user-attachments/assets/3057c66d-e80c-42fa-a41d-61738e33cc09)

**Comment by drjfloyd** [comment]
Support for FDS + EVAC ended in 2022, and EVAC is no longer part of FDS. It was not an easy decision, but one that recognized the reality that the current development team did not have the bandwidth to do the overhaul that would be required to keep it compatible with ongoing FDS development (for example as you have noted issues with parallel processing). You may get lucky and someone who was a user of EVAC and that reads this forum has some advice for you, but you may also want to consider an alternate method of doing an egress calculation -- acquire a Pathfinder or other egress model license or do a hand calcluation following approaches discussed in documents such as the SFPE handbook or NFPA 130.

**Reply by LordMicidial** [reply]
Thank you, Dr. Floyd. Indeed, months ago I understood what you mentioned in the former Google Group for FDS support. I saw a post of yours explaining the decision to discontinue support for EVAC. Based on that, and the fact that the latest EVAC User Manual states that the EVAC module has not been fully validated or verified, I decided to provide hand calculations to the local fire brigade. I might consider switching to Pathfinder for future projects.

For everyone's benefit, could you also confirm whether the EVAC module is no longer under development at VTT or elsewhere? Apart from some publications available online, should we expect any further updates? In the case of evacuation studies, is it advisable to use other tools and approaches instead?

I'll leave the discussion open to see if anyone else would like to contribute.

**Reply by drjfloyd** [reply]
@shostikk do you know if VTT and/or Aalto is doing or has done any work on developing an evacuation model for general use?

FDS + EVAC was last released in version 6.7.7. If you were to use it, it would have to be with the recognition of using an older, no longer supported version of FDS. Since that release there have been 1000s of source code commits. While most are related to improvements / new features, there are a number of bug fixes.

**Reply by johodges** [reply]
The team at Julich and Wuppertal presented at FEMTC 2024 on an open-source pedestrian movement software that they are developing. I have not used this tool personally, but it may be able to do what you want.

https://www.jupedsim.org/stable/

**Reply by LordMicidial** [reply]
> i

Would you mind pointing me to where I can find reliable instructions for hand calculations for EVAC and also for FDS?

**Reply by johodges** [reply]
The SFPE handbook of fire protection engineering is a good starting point on the hand calculations.

Gwynne, Steven MV, and Eric R. Rosenbaum. "Employing the hydraulic model in assessing emergency movement." SFPE handbook of fire protection engineering (2016): 2115-2151.

**Reply by LordMicidial** [reply]
@johodges noted thanks. Any chance you might also suggest a useful read for understanding and perform hand calculations to grasp precisely the fundamental physics of FDS?

At this point, I see I know how how does it work but I'd like to understand the software deeply.

**Comment by drjfloyd** [comment]
If you went from 24 meshes with IJK=120,110,55 to a single mesh with
120,110,55, then it is not surprising that you are seeing substantially
different results. You would have gone from a grid that is 0.2 m x 0.2 m x
0.2 m to one that is 1.2 m x 0.8 m x 0.2 m.  This exceeds the recommended
upper limit on aspect ratio and is a significant reduction in resolution.

**Reply by LordMicidial** [reply]
Yup! Thats why I deleted it after my initial post. It is always a wonder to have the opportunity to engage with top scientists like you, even on Sundays. Thank you!

**Comment by shostikk** [comment]
To my knowledge, FDS+Evac development is not going on at VTT, and definitelly not at Aalto University.

Another free, agent-based evacuation tool is that embedded in [AAMKS](https://github.com/aamks)
